lib/synapse/config_generator/haproxy.rb

```diff
@@ -1319,13 +1322,38 @@ def write_config(new_config)
       if old_config == new_config
         return false
       else
-        tmp_file_path = "#{opts['config_file_path']}.tmp"
-        File.write(tmp_file_path, new_config)
-        FileUtils.mv(tmp_file_path, opts['config_file_path'])
+        candidate_file_path = opts['candidate_config_file_path']
```

Comment:
what if user only set one of them, `candidate_config_file_path` and `check_commands`